File: frontend/src/app/services/owner.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {OwnerModel} from "../main-module/models/owner.model";
import {Content} from "../main-module/models/content";
import {CustomerModel} from "../main-module/models/customer.model";

@Injectable({
    providedIn: 'root'
})
export class OwnerService {

    private _controllerUrlPrefix: string = '/api/owners';

    constructor(private http: HttpClient) {
    }

    getOwners(page: number, size: number): Observable<Content<OwnerModel>> {
        return this.http.get<Content<OwnerModel>>(this._controllerUrlPrefix + '?page=' + page + '&size=' + size);
    }

    updateOwnerDetails(owner: OwnerModel): Observable<CustomerModel> {
        return this.http.put<CustomerModel>(this._controllerUrlPrefix + '/details', owner);
    }

    saveOwner(owner: OwnerModel): Observable<OwnerModel> {
        return this.http.post<OwnerModel>(this._controllerUrlPrefix, owner);
    }

    deleteOwner(id: string): Observable<void> {
        return this.http.delete<void>(this._controllerUrlPrefix + '/' + id);
    }

    getOwnerByUserId(): Observable<OwnerModel> {
        return this.http.get<OwnerModel>(this._controllerUrlPrefix + '/users');
    }
}

File: apigateway/src/main/java/ru/otus/apigateway/service/api/OwnerService.java
package ru.otus.apigateway.service.api;

import ru.otus.apigateway.model.view.Content;
import ru.otus.apigateway.model.view.OwnerViewModel;

import java.util.Optional;

public interface OwnerService {
    Content<OwnerViewModel> getAll(int page, int size);

    Optional<OwnerViewModel> getOwnerById(Long id);

    OwnerViewModel getOwnerByUserId(Long id);

    OwnerViewModel saveOwner(OwnerViewModel owner);

    void saveEditedOwner(OwnerViewModel owner);

    void deleteOwner(Long id);

    void updateOwnerDetails(OwnerViewModel owner);
}
